### PR TITLE
Fix marking/results view panes

### PR DESCRIPTION
**Pull request by arkon**
Replaced the hacky code that replaced the original Prototype-dependent column panes in the marking/results view I wrote last summer with a [new plugin](http://echeung.me/jquery-dualpane/).

EDIT: Also fixes annotations scrolling on images, horizontal scrolling for text files, and improves the old summary toggling.

### CODE REVIEW

**Comment by arkon** [comment]
Hold on, image annotation is kind of borked.

**Comment by david-yz-liu** [comment]
Thanks, @arkon. Two preliminary comments:

1. I don't mind the preview pane always appearing, but it looks like the preview isn't cleared between annotations.
2. PDF submissions didn't render for me (Chrome):

image

File: app/assets/stylesheets/grader.scss [hunk 1]
```diff
@@ -107,47 +107,66 @@
...
 #sel_box {
   border: 1px dashed #f00;
+  cursor: crosshair;
   display: none;
   position: absolute;
 }
 
 .annotation_holder {
-  background: orange;
+  background: #ffa500;
```
Comment (david-yz-liu) [inline app/assets/stylesheets/grader.scss]: @arkon This one seems like a good point to me. Why the change?

File: app/views/results/common/_annotations.js.erb [hunk 2]
```diff
@@ -133,10 +133,10 @@
     var line_end     = line_manager.getLineNumber(focus_node);
 
     // If we selected an entire line the above returns + 1, a fix follows
-    if (mouseSelection.anchorNode.nodeName === "LI"){
+    if (mouseSelection.anchorNode.nodeName === "LI") {
```
Comment (david-yz-liu) [inline app/views/results/common/_annotations.js.erb]: I'm surprised you didn't change to single-quotes here (since you're improving style in this one too)

File: app/views/results/common/_annotations.js.erb [hunk 3]
```diff
@@ -232,16 +232,14 @@
...
-               document.getElementById('image_container').scrollLeft;
+               document.getElementById('image_container').offsetLeft;
     var y1 = parseInt(box.style.top, 10) -
-               document.getElementById('image_preview').offsetTop +
-               document.getElementById('codeviewer').scrollTop;
+               document.getElementById('image_container').offsetTop;
     var x2 = x1 + parseInt(box.style.width, 10);
     var y2 = y1 + parseInt(box.style.height, 10);
 
     if ((x2 - x1) < 1 || isNaN(x2 - x1)) {
```
Comment (david-yz-liu) [inline app/views/results/common/_annotations.js.erb]: Can you change the first part to `x2 - x1 < 1`, please?

File: config/locales/en.yml [hunk 4]
```diff
@@ -344,7 +344,7 @@ en:
         new_annotation: "Create new annotation"
         annotation_category: "Annotation Category"
         preview: "Preview"
-        preview_title: "Here's how the annotation will look:"
+        preview_title: "Preview:"
```
Comment (david-yz-liu) [inline config/locales/en.yml]: Please remove the colon

**Comment by arkon** [comment]
@david-yz-liu Fixed those 2 issues.

File: app/assets/stylesheets/common/_constants.scss [hunk 5]
```diff
@@ -1,24 +1,34 @@
...
+
+// Timings
+$time-quick       : 0.25s;
+$time-medium      : 0.35s;
+
+// Colours
+$lighter-blue  : #e8f4f2;
+$light-blue    : #cee3ea;
+$blue          : #89b1dd;
+$dark-blue     : #3a6ab3
```
Comment (david-yz-liu) [inline app/assets/stylesheets/common/_constants.scss]: Semicolon missing here, and line 26

Comment (arkon) [inline app/assets/stylesheets/common/_constants.scss]: @david-yz-liu Whoops, not sure how I missed that. Fixed.

**Comment by david-yz-liu** [comment]
Thanks! Looks like the PDF correctly displays for Admins and TAs, but not students.

Also, for larger images and text files, the vertical scrollbar appears for the whole marking/annotation pane, rather than just the image itself. This means when you scroll down, you actually lose sight of the top tabs, which isn't great. On the other hand, this does *not* happen for PDFs.

One nice thing is you seem you have fixed the horizontal scrolling of #1809, though correct me if I'm wrong!

**Comment by arkon** [comment]
@david-yz-liu Weird, PDFs seem to work as a student for me. Mind posting a screenshot or something of what you see (or is just the same as the previous screenshot)?

I did make the code have line breaks so the lines wrap instead of having to scroll instead.

I'll fiddle around more with images and text files.

**Comment by arkon** [comment]
@david-yz-liu Both text files and images should scroll properly now.

**Comment by david-yz-liu** [comment]
@arkon Regarding the PDFs: in the instructor view, *after* releasing an assignment, I also do not have the PDF appearing (same blank div as above). I opened the console, and get:

````
TypeError: document.getElementById(...) is null
document.getElementById('old_summary_criteria_pane').style.display = 'none';
edit (line 296, col 4)
```

Perhaps this is an unrelated issue, but it's worth checking out anyway.

**Comment by arkon** [comment]
@david-yz-liu I can't seem to reproduce it, but I tweaked the old summary criteria pane toggling function. It's also a bit better now since there's a button acting as a toggle instead of relying on mousing over an anchor link to show the old marks.

**Comment by david-yz-liu** [comment]
@arkon My bad, I think the PDF was just slow to load on my computer, and I needed to scroll down to see the loading icon.

**Comment by david-yz-liu** [comment]
For the scrolling, I am referring to this: 

scroll

I would like the scroll bar to not scroll away the "Source Code" and "Annotations" tab area.

**Comment by arkon** [comment]
image

I thought I fixed that?

**Comment by david-yz-liu** [comment]
I'm on Firefox.

**Comment by arkon** [comment]
@david-yz-liu Ah okay, it's broken for me in Firefox too. I'll take a look at it.

**Comment by arkon** [comment]
I'm struggling to get it working in both Chrome and Firefox. What I have right now seems to be the best, I had it working in Firefox at one point but then it was wildly broken in Chrome instead.

**Comment by david-yz-liu** [comment]
@arkon Thanks for looking into this. I'll check it out tomorrow.
